pallets/passkey/src/tests.rs

```diff
@@ -319,31 +317,6 @@ fn fee_withdrawn_for_failed_call() {
 	});
 }
 
-#[test]
-fn fee_not_removed_on_failed_validation() {
```

Comment:
We don't need this test because the fees would be deducted, but it is the responsibility of the caller to reverse the changes in storage based on docs.

Reply:
Would you mind linking to the documentation you're referring to?

Reply:
So will it be a batch call ?

Reply:
@shannonwells https://paritytech.github.io/polkadot-sdk/master/polkadot_service/runtime_traits/trait.ValidateUnsigned.html#method.pre_dispatch

Reply:
I'm not sure if I understand what you mean by batch call. So my understanding is that the pre_dispatch caller which is inside executor or frame (I don't remember which one) checks for the result of pre_dispatch and only persists the storage changes if the result was OK. More details how it does is probably since all the writes go into the caching layer. it probably cleans the caching layer in case of any error so it does not get persisted at the end of block

Reply:
The link above solved my query, thanks